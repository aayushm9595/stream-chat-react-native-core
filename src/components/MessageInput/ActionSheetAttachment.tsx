import React from 'react';
import { ActionSheetCustom } from 'react-native-actionsheet';

import type {
  ImageRequireSource,
  StyleProp,
  TextStyle,
  ViewStyle,
} from 'react-native';

import { IconSquare } from '../IconSquare';

import { useTranslationContext } from '../../contexts/translationContext/TranslationContext';

import { styled } from '../../styles/styledComponents';

const iconGallery: ImageRequireSource = require('../../images/icons/icon_attach-media.png');
const iconClose: ImageRequireSource = require('../../images/icons/icon_close.png');
const iconFolder: ImageRequireSource = require('../../images/icons/icon_folder.png');

const ActionSheetButtonContainer = styled.View`
  align-items: center;
  flex-direction: row;
  justify-content: flex-start;
  padding-left: 20px;
  width: 100%;
  ${({ theme }) => theme.messageInput.actionSheet.buttonContainer.css};
`;

const ActionSheetButtonText = styled.Text`
  ${({ theme }) => theme.messageInput.actionSheet.buttonText.css};
`;

const ActionSheetTitleContainer = styled.View`
  align-items: center;
  flex-direction: row;
  height: 100%;
  justify-content: space-between;
  padding-horizontal: 20px;
  width: 100%;
  ${({ theme }) => theme.messageInput.actionSheet.titleContainer.css};
`;

const ActionSheetTitleText = styled.Text`
  font-weight: bold;
  ${({ theme }) => theme.messageInput.actionSheet.titleText.css};
`;

type AttachmentActionSheetItemProps = {
  icon: ImageRequireSource;
  key: string | number;
  text: string;
  testID?: string;
};

const AttachmentActionSheetItem: React.FC<AttachmentActionSheetItemProps> = (
  props,
) => {
  const { icon, key, testID, text } = props;
  return (
    <ActionSheetButtonContainer key={key} testID={testID}>
      <IconSquare icon={icon} />
      <ActionSheetButtonText>{text}</ActionSheetButtonText>
    </ActionSheetButtonContainer>
  );
};

export type ActionSheetStyles = {
  body?: StyleProp<ViewStyle>;
  buttonBox?: StyleProp<ViewStyle>;
  buttonText?: StyleProp<TextStyle>;
  cancelButtonBox?: StyleProp<ViewStyle>;
  messageBox?: StyleProp<ViewStyle>;
  messageText?: StyleProp<TextStyle>;
  overlay?: StyleProp<TextStyle>;
  titleBox?: StyleProp<ViewStyle>;
  titleText?: StyleProp<TextStyle>;
  wrapper?: StyleProp<ViewStyle>;
};

export type ActionSheetProps = {
  closeAttachActionSheet: () => void;
  pickFile: () => Promise<void>;
  pickImage: () => Promise<void>;
  setAttachActionSheetRef: (ref: ActionSheetCustom | null) => void;
  styles?: ActionSheetStyles;
};

export const ActionSheetAttachment: React.FC<ActionSheetProps> = (props) => {
  const {
    closeAttachActionSheet,
    pickFile,
    pickImage,
    setAttachActionSheetRef,
    styles,
  } = props;

  const { t } = useTranslationContext();

  return (
    <ActionSheetCustom
      onPress={(index) => {
        // https://github.com/beefe/react-native-actionsheet/issues/36
        setTimeout(() => {
          switch (index) {
            case 0:
              pickImage();
              break;
            case 1:
              pickFile();
              break;
            default:
          }
        }, 201); // 201ms to fire after the animation is complete https://github.com/beefe/react-native-actionsheet/blob/master/lib/ActionSheetCustom.js#L78
      }}
      options={[
        <AttachmentActionSheetItem
          icon={iconGallery}
          key='upload-photo-item'
          testID='upload-photo-item'
          text={t('Upload a photo')}
        />,
        <AttachmentActionSheetItem
          icon={iconFolder}
          key='upload-file-item'
          testID='upload-file-item'
          text={t('Upload a file')}
        />,
      ]}
      ref={setAttachActionSheetRef}
      styles={styles}
      title={
        <ActionSheetTitleContainer>
          <ActionSheetTitleText>{t('Add a file')}</ActionSheetTitleText>
          <IconSquare icon={iconClose} onPress={closeAttachActionSheet} />
        </ActionSheetTitleContainer>
      }
    />
  );
};
