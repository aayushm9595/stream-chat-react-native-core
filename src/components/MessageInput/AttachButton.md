Attach button

```js
import { AttachButton } from './AttachButton';

<AttachButton />
```

Disabled Attach button

```js
import { AttachButton } from './AttachButton';

<AttachButton disabled />
```
