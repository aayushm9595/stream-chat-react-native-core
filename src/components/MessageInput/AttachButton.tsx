import React from 'react';

import type { GestureResponderEvent, ImageRequireSource } from 'react-native';

import { styled } from '../../styles/styledComponents';

const iconAddAttachment: ImageRequireSource = require('../../images/icons/plus-outline.png');

const AttachButtonIcon = styled.Image`
  height: 15px;
  width: 15px;
  ${({ theme }) => theme.messageInput.attachButtonIcon.css};
`;

const Container = styled.TouchableOpacity`
  margin-right: 8px;
  ${({ theme }) => theme.messageInput.attachButton.css};
`;

export type AttachButtonProps = {
  /** Disables the button */
  disabled?: boolean;
  /** Function that opens an attachment action sheet */
  handleOnPress?: (event: GestureResponderEvent) => void;
};

/**
 * UI Component for attach button in MessageInput component.
 *
 * @example ./AttachButton.md
 */
export const AttachButton: React.FC<AttachButtonProps> = (props) => {
  const { disabled = false, handleOnPress } = props;
  return (
    <Container
      disabled={disabled}
      onPress={handleOnPress}
      testID='attach-button'
    >
      <AttachButtonIcon source={iconAddAttachment} />
    </Container>
  );
};
