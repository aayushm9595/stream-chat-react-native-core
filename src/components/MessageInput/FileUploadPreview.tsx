import React from 'react';
import { FlatList, ImageRequireSource } from 'react-native';

import { UploadProgressIndicator } from './UploadProgressIndicator';

import type { FileUpload } from './hooks/useMessageDetailsForState';

import { FileIcon, FileIconProps } from '../Attachment/FileIcon';

import { styled } from '../../styles/styledComponents';
import { FileState, ProgressIndicatorTypes } from '../../utils/utils';

const closeRound: ImageRequireSource = require('../../images/icons/close-round.png');

const FILE_PREVIEW_HEIGHT = 50;
const FILE_PREVIEW_PADDING = 10;

const AttachmentContainerView = styled.View`
  align-items: center;
  border-color: #ebebeb;
  border-width: 0.5px;
  flex-direction: row;
  height: ${FILE_PREVIEW_HEIGHT}px;
  justify-content: space-between;
  margin-bottom: 5px;
  padding: ${FILE_PREVIEW_PADDING}px;
  ${({ theme }) =>
    theme.messageInput.fileUploadPreview.attachmentContainerView.css};
`;

const AttachmentView = styled.View`
  align-items: center;
  flex-direction: row;
  ${({ theme }) => theme.messageInput.fileUploadPreview.attachmentView.css};
`;

const Container = styled.View<{ fileUploadsLength: number }>`
  height: ${({ fileUploadsLength }) =>
    fileUploadsLength * (FILE_PREVIEW_HEIGHT + 5)}px;
  margin-horizontal: 10px;
  ${({ theme }) => theme.messageInput.fileUploadPreview.container.css};
`;

const Dismiss = styled.TouchableOpacity`
  align-items: center;
  background-color: #fff;
  border-radius: 20px;
  height: 20px;
  justify-content: center;
  position: absolute;
  right: 5px;
  top: 5px;
  width: 20px;
  ${({ theme }) => theme.messageInput.fileUploadPreview.dismiss.css};
`;

const DismissImage = styled.Image`
  height: 10px;
  width: 10px;
  ${({ theme }) => theme.messageInput.fileUploadPreview.dismissImage.css};
`;

const FilenameText = styled.Text`
  padding-left: 10px;
  ${({ theme }) => theme.messageInput.fileUploadPreview.filenameText.css};
`;

export type FileUploadPreviewProps = {
  /**
   * An array of file objects which are set for upload. It has the following structure:
   *
   * ```json
   *  [
   *    {
   *      "file": // File object,
   *      "id": "randomly_generated_temp_id_1",
   *      "state": "uploading" // or "finished",
   *      "url": "https://url1.com",
   *    },
   *    {
   *      "file": // File object,
   *      "id": "randomly_generated_temp_id_2",
   *      "state": "uploading" // or "finished",
   *      "url": "https://url1.com",
   *    },
   *  ]
   * ```
   *
   */
  fileUploads: FileUpload[];
  /**
   * Function for removing a file from the upload preview
   *
   * @param id string ID of file in `fileUploads` object in state of MessageInput
   */
  removeFile: (id: string) => void;
  /**
   * Function for attempting to upload a file
   *
   * @param id string ID of file in `fileUploads` object in state of MessageInput
   */
  retryUpload: ({ newFile }: { newFile: FileUpload }) => Promise<void>;
  /**
   * Custom UI component for attachment icon for type 'file' attachment.
   * Defaults to and accepts same props as: https://github.com/GetStream/stream-chat-react-native/blob/master/src/components/Attachment/FileIcon.tsx
   */
  AttachmentFileIcon?: React.ComponentType<FileIconProps>;
};

/**
 * FileUploadPreview
 * UI Component to preview the files set for upload
 *
 * @example ./FileUploadPreview.md
 */
export const FileUploadPreview: React.FC<FileUploadPreviewProps> = (props) => {
  const {
    AttachmentFileIcon = FileIcon,
    fileUploads,
    removeFile,
    retryUpload,
  } = props;

  const renderItem = ({ item }: { item: FileUpload }) => (
    <>
      <UploadProgressIndicator
        action={() => {
          if (retryUpload) {
            retryUpload({ newFile: item });
          }
        }}
        active={item.state !== FileState.UPLOADED}
        type={
          item.state === FileState.UPLOADING
            ? ProgressIndicatorTypes.IN_PROGRESS
            : item.state === FileState.UPLOAD_FAILED
            ? ProgressIndicatorTypes.RETRY
            : undefined
        }
      >
        <AttachmentContainerView>
          <AttachmentView>
            <AttachmentFileIcon mimeType={item.file.type} size={20} />
            <FilenameText>
              {item.file.name
                ? item.file.name.length > 35
                  ? item.file.name.substring(0, 35).concat('...')
                  : item.file.name
                : ''}
            </FilenameText>
          </AttachmentView>
        </AttachmentContainerView>
      </UploadProgressIndicator>
      <Dismiss
        onPress={() => {
          if (removeFile) {
            removeFile(item.id);
          }
        }}
        testID='remove-file-upload-preview'
      >
        <DismissImage source={closeRound} />
      </Dismiss>
    </>
  );

  return fileUploads?.length > 0 ? (
    <Container fileUploadsLength={fileUploads.length}>
      <FlatList
        data={fileUploads}
        keyExtractor={(item) => item.id}
        renderItem={renderItem}
        style={{ flex: 1 }}
      />
    </Container>
  ) : null;
};
