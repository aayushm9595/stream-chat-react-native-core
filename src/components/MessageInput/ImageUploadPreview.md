```js
import React from 'react';

import { ImageUploadPreview } from './ImageUploadPreview';

const imageUploads = [
  {
    id: '1',
    file: {
      uri: 'https://picsum.photos/200/300?random=1',
    },
  },
  {
    id: '2',
    file: {
      uri: 'https://picsum.photos/200/300?random=2',
    },
  },
  {
    id: '3',
    file: {
      uri: 'https://picsum.photos/200/300?random=3',
    },
  },
];

const IUPExample = () => {
  const [imageUploadsState, setImageUploadsState] = React.useState(imageUploads);

  const removeImage = (id) => {
    setImageUploadsState((prevImageUploads) => prevImageUploads.filter((image) => image.id !== id));
  };

  return (
    <ImageUploadPreview
      imageUploads={imageUploadsState}
      removeImage={removeImage}
    />
  );
};

<IUPExample />;
```
