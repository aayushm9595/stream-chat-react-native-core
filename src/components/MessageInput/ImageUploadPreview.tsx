import React from 'react';
import { FlatList, ImageRequireSource } from 'react-native';

import { UploadProgressIndicator } from './UploadProgressIndicator';

import type { ImageUpload } from './hooks/useMessageDetailsForState';

import { styled } from '../../styles/styledComponents';
import { FileState, ProgressIndicatorTypes } from '../../utils/utils';

const closeRound: ImageRequireSource = require('../../images/icons/close-round.png');

const Container = styled.View`
  height: 70px;
  padding: 10px;
  ${({ theme }) => theme.messageInput.imageUploadPreview.container.css};
`;

const Dismiss = styled.TouchableOpacity`
  align-items: center;
  background-color: #fff;
  border-radius: 20px;
  height: 20px;
  justify-content: center;
  position: absolute;
  right: 5px;
  top: 5px;
  width: 20px;
  ${({ theme }) => theme.messageInput.imageUploadPreview.dismiss.css};
`;

const DismissImage = styled.Image`
  height: 10px;
  width: 10px;
  ${({ theme }) => theme.messageInput.imageUploadPreview.dismissImage.css};
`;

const ItemContainer = styled.View`
  align-items: flex-start;
  flex-direction: row;
  height: 50px;
  margin-left: 5px;
  ${({ theme }) => theme.messageInput.imageUploadPreview.itemContainer.css};
`;

const Upload = styled.Image`
  border-radius: 10px;
  height: 50px;
  width: 50px;
  ${({ theme }) => theme.messageInput.imageUploadPreview.upload.css};
`;

export type ImageUploadPreviewProps = {
  /**
   * An array of image objects which are set for upload. It has the following structure:
   *
   * ```json
   *  [
   *    {
   *      "file": // File object,
   *      "id": "randomly_generated_temp_id_1",
   *      "state": "uploading" // or "finished",
   *    },
   *    {
   *      "file": // File object,
   *      "id": "randomly_generated_temp_id_2",
   *      "state": "uploading" // or "finished",
   *    },
   *  ]
   * ```
   *
   */
  imageUploads: ImageUpload[];
  /**
   * Function for removing an image from the upload preview
   *
   * @param id string ID of image in `imageUploads` object in state of MessageInput
   */
  removeImage: (id: string) => void;
  /**
   * Function for attempting to upload an image
   *
   * @param id string ID of image in `imageUploads` object in state of MessageInput
   */
  retryUpload: ({ newImage }: { newImage: ImageUpload }) => Promise<void>;
};

/**
 * UI Component to preview the images set for upload
 *
 * @example ./ImageUploadPreview.md
 */
export const ImageUploadPreview: React.FC<ImageUploadPreviewProps> = (
  props,
) => {
  const { imageUploads, removeImage, retryUpload } = props;

  const renderItem = ({ item }: { item: ImageUpload }) => (
    <ItemContainer>
      <UploadProgressIndicator
        action={() => {
          if (retryUpload) {
            retryUpload({ newImage: item });
          }
        }}
        active={item.state !== FileState.UPLOADED}
        type={
          item.state === FileState.UPLOADING
            ? ProgressIndicatorTypes.IN_PROGRESS
            : item.state === FileState.UPLOAD_FAILED
            ? ProgressIndicatorTypes.RETRY
            : undefined
        }
      >
        <Upload
          resizeMode='cover'
          source={{ uri: item.file.uri || item.url }}
        />
      </UploadProgressIndicator>
      <Dismiss
        onPress={() => {
          if (removeImage) {
            removeImage(item.id);
          }
        }}
        testID='remove-image-upload-preview'
      >
        <DismissImage source={closeRound} />
      </Dismiss>
    </ItemContainer>
  );

  return imageUploads?.length > 0 ? (
    <Container>
      <FlatList
        data={imageUploads}
        horizontal
        keyExtractor={(item) => item.id}
        renderItem={renderItem}
        style={{ flex: 1 }}
      />
    </Container>
  ) : null;
};
