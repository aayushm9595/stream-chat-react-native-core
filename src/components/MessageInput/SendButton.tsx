import React from 'react';

import { useMessagesContext } from '../../contexts/messagesContext/MessagesContext';
import { styled } from '../../styles/styledComponents';

import type { GestureResponderEvent, ImageRequireSource } from 'react-native';

import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../types/types';

const iconEdit: ImageRequireSource = require('../../images/icons/icon_edit.png');
const iconSendNewMessage: ImageRequireSource = require('../../images/icons/icon_new_message.png');

const Container = styled.TouchableOpacity`
  margin-left: 8px;
  ${({ theme }) => theme.messageInput.sendButton.css};
`;

const SendButtonIcon = styled.Image`
  height: 15px;
  width: 15px;
  ${({ theme }) => theme.messageInput.sendButtonIcon.css};
`;

export type SendButtonProps = {
  /** Disables the button */
  disabled?: boolean;
  /** Function that sends message */
  sendMessage?: (event: GestureResponderEvent) => void;
};

/**
 * UI Component for send button in MessageInput component.
 *
 * @example ./SendButton.md
 */
export const SendButton = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  props: SendButtonProps,
) => {
  const { disabled = false, sendMessage } = props;
  const { editing } = useMessagesContext<At, Ch, Co, Ev, Me, Re, Us>();
  return (
    <Container disabled={disabled} onPress={sendMessage} testID='send-button'>
      <SendButtonIcon source={editing ? iconEdit : iconSendNewMessage} />
    </Container>
  );
};
