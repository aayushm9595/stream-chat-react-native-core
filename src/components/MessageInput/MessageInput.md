```js
import { MessageInput } from './MessageInput';

import { Channel } from '../Channel/Channel';
import { Chat } from '../Chat/Chat';
import { channel, client } from '../docs/data';

<Chat client={client}>
    <Channel channel={channel}>
        <MessageInput />
    </Channel>
</Chat>
```