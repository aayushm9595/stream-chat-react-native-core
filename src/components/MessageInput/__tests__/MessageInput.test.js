import React from 'react';
import {
  cleanup,
  fireEvent,
  render,
  waitFor,
} from '@testing-library/react-native';

import { getOrCreateChannelApi } from 'mock-builders/api/getOrCreateChannel';
import { useMockedApis } from 'mock-builders/api/useMockedApis';
import { generateChannel } from 'mock-builders/generator/channel';
import { generateUser } from 'mock-builders/generator/user';
import { getTestClientWithUser } from 'mock-builders/mock';

import { MessageInput } from '../MessageInput';

import { Channel } from '../../Channel/Channel';
import { Chat } from '../../Chat/Chat';

describe('MessageInput', () => {
  const clientUser = generateUser();
  let chatClient;
  let channel;

  const getComponent = () => (
    <Chat client={chatClient}>
      <Channel channel={channel}>
        <MessageInput />
      </Channel>
    </Chat>
  );

  const initializeChannel = async (c) => {
    useMockedApis(chatClient, [getOrCreateChannelApi(c)]);

    channel = chatClient.channel('messaging');

    await channel.watch();
  };

  beforeEach(async () => {
    chatClient = await getTestClientWithUser(clientUser);
  });

  afterEach(() => {
    channel = null;
    cleanup();
  });

  it('should render MessageInput', async () => {
    await initializeChannel(generateChannel());

    const { getByTestId, queryByTestId, queryByText, toJSON } = render(
      getComponent(),
    );

    await waitFor(() => {
      expect(queryByTestId('upload-photo-item')).toBeTruthy();
      expect(queryByTestId('upload-file-item')).toBeTruthy();
      expect(queryByTestId('attach-button')).toBeTruthy();
      expect(queryByTestId('auto-complete-text-input')).toBeTruthy();
      expect(queryByTestId('send-button')).toBeTruthy();
      expect(queryByText('Editing Message')).toBeFalsy();
    });

    fireEvent.press(getByTestId('attach-button'));

    const snapshot = toJSON();

    await waitFor(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
