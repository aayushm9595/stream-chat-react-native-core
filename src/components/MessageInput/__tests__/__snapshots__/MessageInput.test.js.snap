// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput should render MessageInput 1`] = `
<View
  keyboardVerticalOffset={86.5}
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      imageUploads={Array []}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.05)",
            "borderRadius": 10,
            "marginHorizontal": 10,
            "paddingTop": 0,
          },
        ]
      }
    >
      <Modal
        animationType="none"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Array [
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                },
              ],
            ]
          }
        >
          <Text
            onPress={[Function]}
            style={
              Array [
                Array [
                  Object {
                    "backgroundColor": "#000",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0.4,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ],
              ]
            }
          />
          <View
            style={
              Object {
                "alignSelf": "flex-end",
                "backgroundColor": "#e5e5e5",
                "flex": 1,
                "height": 141,
                "transform": Array [
                  Object {
                    "translateY": 141,
                  },
                ],
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "height": 40,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": "100%",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 20,
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontWeight": "bold",
                      },
                    ]
                  }
                >
                  Add a file
                </Text>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.05)",
                      "borderRadius": 5,
                      "opacity": 1,
                      "paddingBottom": 5,
                      "paddingLeft": 5,
                      "paddingRight": 5,
                      "paddingTop": 5,
                    }
                  }
                  testID="icon-square"
                >
                  <Image
                    source={
                      Object {
                        "testUri": "../../../src/images/icons/icon_close.png",
                      }
                    }
                    style={
                      Object {
                        "height": 15,
                        "width": 15,
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <RCTScrollView
              scrollEnabled={false}
            >
              <View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#fff",
                        "height": 50,
                        "justifyContent": "center",
                        "marginTop": 0.5,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "flex-start",
                          "paddingLeft": 20,
                          "width": "100%",
                        },
                      ]
                    }
                    testID="upload-photo-item"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.05)",
                            "borderRadius": 5,
                            "paddingBottom": 5,
                            "paddingLeft": 5,
                            "paddingRight": 5,
                            "paddingTop": 5,
                          },
                        ]
                      }
                      testID="icon-square"
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../src/images/icons/icon_attach-media.png",
                          }
                        }
                        style={
                          Object {
                            "height": 15,
                            "width": 15,
                          }
                        }
                      />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {},
                        ]
                      }
                    >
                      Upload a photo
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#fff",
                        "height": 50,
                        "justifyContent": "center",
                        "marginTop": 0.5,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "flex-start",
                          "paddingLeft": 20,
                          "width": "100%",
                        },
                      ]
                    }
                    testID="upload-file-item"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.05)",
                            "borderRadius": 5,
                            "paddingBottom": 5,
                            "paddingLeft": 5,
                            "paddingRight": 5,
                            "paddingTop": 5,
                          },
                        ]
                      }
                      testID="icon-square"
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../src/images/icons/icon_folder.png",
                          }
                        }
                        style={
                          Object {
                            "height": 15,
                            "width": 15,
                          }
                        }
                      />
                    </View>
                    <Text
                      style={
                        Array [
                          Object {},
                        ]
                      }
                    >
                      Upload a file
                    </Text>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </Modal>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "minHeight": 46,
              "paddingHorizontal": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginRight": 8,
              "opacity": 1,
            }
          }
          testID="attach-button"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/icons/plus-outline.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 15,
                  "width": 15,
                },
              ]
            }
          />
        </View>
        <TextInput
          allowFontScaling={true}
          multiline={true}
          onChangeText={[Function]}
          onSelectionChange={[Function]}
          placeholder="Write your message"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": -5,
                "marginLeft": -5,
                "marginRight": -5,
                "marginTop": -5,
                "maxHeight": 60,
              },
            ]
          }
          testID="auto-complete-text-input"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginLeft": 8,
              "opacity": 1,
            }
          }
          testID="send-button"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/images/icons/icon_new_message.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 15,
                  "width": 15,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
